crates/utils/re_arrow_util/src/lib.rs

```diff
@@ -384,3 +384,87 @@ where
     array.shrink_to_fit(); // VERY IMPORTANT! https://github.com/rerun-io/rerun/issues/7222
     array
 }
+
+// ----------------------------------------------------------------------------
+
+/// Convert `[A, B, null, D, …]` into `[[A], [B], null, [D], …]`
+pub fn wrap_in_list_array(field: &Field, array: ArrayRef) -> (Field, ListArray) {
+    re_tracing::profile_function!();
+
+    debug_assert_eq!(field.data_type(), array.data_type());
+
+    let item_field = Arc::new(Field::new(
+        "item",
+        field.data_type().clone(),
+        field.is_nullable(), // TODO(emilk): it would be nice to remove the "inner nullability", and just have outer nullability.
+    ));
+
+    let offsets = OffsetBuffer::from_lengths(std::iter::repeat(1).take(array.len()));
+    let nulls = array.nulls().cloned();
+    let list_array = ListArray::new(item_field, offsets, array, nulls);
+
+    let list_field = Field::new(
+        field.name().clone(),
+        list_array.data_type().clone(),
+        true, // All components in Rerun has "outer nullability"
+    )
+    .with_metadata(field.metadata().clone());
+
+    (list_field, list_array)
+}
+
+#[cfg(test)]
+mod tests {
+
+    use arrow::{
+        array::{Array as _, AsArray as _, Int32Array},
+        buffer::{NullBuffer, ScalarBuffer},
+        datatypes::{DataType, Int32Type},
+    };
+
+    use super::*;
+
+    #[test]
+    fn test_wrap_in_list_array() {
+        // Convert [42, 1337, null, 69] into [[42], [1337], null, [69]]
```

Comment:
```suggestion
        // Convert [42, 69, null, 1337] into [[42], [69], null, [1337]]
```
Inconsistent with the data below.